docs/analytics-ga4/ga4-link-tracker.md

```diff
@@ -1,116 +1,75 @@
-- Passing a `dedicatedDownloadDomains` array allows you to specify domains which will always be classed as a download link. For example, `dropbox.com` could be added to always track dropbox links as download links.
-- Passing `disableListeners: true` prevents click listeners from being added to the page on initilisation. This is useful if you'd like use the link detection features elsewhere without adding extra click listeners to the DOM.
+# Google Analytics 4 link tracker
 
-```JavaScript
-//= require ./analytics-ga4/ga4-link-tracker
+This script is intended for adding GA4 tracking to links. It depends upon the main GA4 analytics code to function.
 
-window.GOVUK.analyticsGa4.linkTracker.init({
-    internalDomains: ['www.gov.uk'],
-    internalDownloadPaths: ['/government/uploads/', '/downloads/'],
-    dedicatedDownloadDomains: ['assets.publishing.service.gov.uk', 'dropbox.com'],
-    disableListeners: // true/false
-})
+## Basic use (single link)
 
+```html
+<a
+  href="/link"
+  data-module="ga4-track-click"
+  data-ga4-link='{ "event_name": "navigation", "type": "home page", "index": 0, "index_total": 1, "section": "name of section", "external": "false" }'>
+    Link
+</a>
 ```
 
-`<a href="https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/742746/A_quick_guide_to_govt_healthy_eating_update.pdf">A Quick Guide to the Government’s Healthy Eating Recommendations</a>`
-
-In the example above, on a left click of the link, the following would be pushed to the dataLayer, using the `eventSchema` found in `ga4-schemas.js`:
-
-```JSON
-{
-    "event": "event_data",
-    "event_data": {
-        "event_name": "file_download",
-        "type": "generic download",
-        "url": "https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/742746/A_quick_guide_to_govt_healthy_eating_update.pdf",
-        "text": "A Quick Guide to the Government’s Healthy Eating Recommendations",
-        "index": null,
-        "index_total": null,
-        "section": null,
-        "action": null,
-        "external": "true",
-        "method": "primary click",
-        "link_domain": "https://assets.publishing.service.gov.uk",
-        "link_path_parts": {
-            "1": "/government/uploads/system/uploads/attachment_data/file/742746/A_quick_guide_to_govt_healthy_eating_",
-            "2": "update.pdf"
-        }
-    }
-}
+Note that the specific detail of the `data-ga4-link` attribute will depend on the context of the link.
+
+## Basic use (multiple links)
+
+Specific tracking can be applied to multiple elements within a container, by applying the data module once to the parent element.
+
+```html
+<div data-module="ga4-track-click">
+  <a
+    href="/a-page"
+    data-ga4-link='{ "event_name": "navigation", "type": "browse", "index": "0", "index_total": "2", "section": "name of section", "external": "false" }'>
```

Comment:
Remove external here as any value in the JSON gets overwritten by the code now.

Reply:
Removed, thanks.